network/cmd/cmd.go

```diff
@@ -49,9 +49,9 @@ func FlagSet() *pflag.FlagSet {
 		"Required when `network.enabletls` is `true`.")
 	flagSet.String("network.truststorefile", defs.TrustStoreFile, "PEM file containing the trusted CA certificates for authenticating remote gRPC servers.")
 	flagSet.String("network.nodedid", defs.NodeDID, "Specifies the DID of the organization that operates this node, typically a vendor for EPD software. It is used to identify the node on the network. If the DID document does not exist of is deactivated, the node will not start.")
-	flagSet.String("network.nats.hostname", defs.Nats.Hostname, "Hostname for the NATS server")
-	flagSet.Int("network.nats.port", defs.Nats.Port, "Port where the NATS server listens on")
-	flagSet.Int("network.nats.timeout", defs.Nats.Timeout, "Timeout (in seconds) for the NATS connection")
+	flagSet.String("network.nats.hostname", defs.ProtocolV2.Nats.Hostname, "Hostname for the NATS server")
```

Comment:
Not sure what you mean exactly but note that you can also configure an external NATS instance

Reply:
They are. If not, how would you be able to connect to an external NATS instance?

Reply:
I think I would prefer to have all NATS config in the `Event Bus` module. Then inject the queue/stream/whatever into the specific modules. Maybe based on an interface the module can implement, which is then queried by the `System` (or maybe the `Event` module) when bootstrapping the Nuts node.

Then the `Event Bus` can be configured to redirect certain topics to an external NATS instance (future work).

E.g.:
```go
type EventPublisher interface {
  GetPublishTopics() []string
  InjectPublishers(topicPublishers map[string]Publisher)
}
```
(and having another one being `EventConsumer`, or make a composite one).

Or maybe better:

```go
type Eventable interface {
   StartPubSub(pubSubber) // called by System/Event Bus, allows engines to start listening on/publishing to topics.
}
```
(requires some better naming)

Reply:
In my opinion the events engine should be optional. If it's disabled you have to use an external NATS instance. Adding yet another tight coupled integration to the engines is not a good idea in my opinion. It quickly becomes a mess because it's really hard to find the right abstraction.